### PR TITLE
doc/refactor(asgi): Start fleshing out docstrings, tweak doc refs, remove add_error_handler shimming

### PR DESCRIPTION
**Pull request by kgriffs**
This patch is a bit of a grab bag of items that I noticed and
fixed while starting work on fleshing out the docs for the new ASGI
implementation.

Note that there is still quite a bit of work to do in order to integrate
ASGI into the docs. Many docs need to be combed and updated to work with
ASGI and WSGI side-by-side, and there are still plenty of missing docs for
the new ASGI classes and functions.

### CODE REVIEW

File: docs/api/errors.rst [hunk 1]
```diff
@@ -63,7 +63,7 @@ All classes are available directly in the ``falcon`` package namespace::
 Note also that any exception (not just instances of
 :class:`~.HTTPError`) can be caught, logged, and otherwise handled
 at the global level by registering one or more custom error handlers.
-See also :meth:`~.App.add_error_handler` to learn more about this
+See also :meth:`~.falcon.App.add_error_handler` to learn more about this
```
Comment (kgriffs) [inline docs/api/errors.rst]: This gets sphinx to build without warnings; for future PRs I need to think about how to better present ASGI and WSGI references side-by-side, as needed.

File: falcon/app.py [hunk 2]
```diff
@@ -238,32 +240,6 @@ def __init__(self, media_type=DEFAULT_MEDIA_TYPE,
         self.add_error_handler(falcon.HTTPError, self._http_error_handler)
         self.add_error_handler(falcon.HTTPStatus, self._http_status_handler)
 
-    def add_middleware(self, middleware):
```
Comment (kgriffs) [inline falcon/app.py]: I reordered a few methods to make them easier to find when going back and forth between `falcon.App` and `falcon.asgi.App`.

File: falcon/asgi/app.py [hunk 3]
```diff
@@ -515,34 +563,114 @@ def __init__(self, *args, request_type=Request, response_type=Response, **kwargs
...
+                        async def handle(req, resp, ex, params):
+                            # TODO: Log the error
+                            # Convert to an instance of falcon.HTTPError
+                            raise falcon.HTTPError(falcon.HTTP_792)
+
+                If an iterable of exception types is specified instead of
+                a single type, the handler must be explicitly specified.
+        """
+
+        if handler is None:
```
Comment (kgriffs) [inline falcon/asgi/app.py]: I went ahead and cleaned up the method implementation here; trying to use super() just made things too convoluted, and there is no reason to support deprecated handler signatures in the new ASGI implementation.

File: falcon/request.py [hunk 4]
```diff
@@ -1865,11 +1866,13 @@ class RequestOptions:
             certain cases, such as when working with authentication
             schemes that employ URL-based signatures.
 
-        default_media_type (str): The default media-type to use when
-            deserializing a response. This value is normally set to the media
-            type provided when a :class:`falcon.App` is initialized; however,
-            if created independently, this will default to the
-            ``DEFAULT_MEDIA_TYPE`` specified by Falcon.
+        default_media_type (str): The default media-type used to
```
Comment (kgriffs) [inline falcon/request.py]: While updating the references to include ASGI, I took the liberty of adding additional clarity to these descriptions.

File: tests/test_error_handlers.py [hunk 5]
```diff
@@ -211,22 +215,6 @@ def legacy_handler3(err, rq, rs, prms):
         client.simulate_get()
         client.simulate_head()
 
-    def test_handler_signature_shim_asgi(self):
```
Comment (kgriffs) [inline tests/test_error_handlers.py]: If someone is porting an old WSGI app over to ASGI, IMHO it is fair to require them to not use deprecated error handler signatures.

File: falcon/asgi/app.py [hunk 6]
```diff
@@ -51,17 +51,27 @@
 # TODO(kgriffs): Rename the WSGI class to App with an API alias kept for
```
Comment (vytas7) [inline falcon/asgi/app.py]: Now that https://github.com/falconry/falcon/issues/1579 is closed, we could revise this TODO :arrow_up:

Comment (kgriffs) [reply falcon/asgi/app.py]: done

**Pull request merged**